File: app/src/main/java/com/example/e_commercemobapp/admin/AdminCategoryAdapter.java
package com.example.e_commercemobapp.admin;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import android.content.Intent;

import com.example.e_commercemobapp.R;

public class AdminCategoryAdapter extends RecyclerView.Adapter<AdminCategoryAdapter.CategoryViewHolder> {

    private List<AdminCategory> categoryList;

    public AdminCategoryAdapter(List<AdminCategory> categoryList) {
        this.categoryList = categoryList;
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.admin_item_category, parent, false);
        return new CategoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder holder, int position) {
        AdminCategory category = categoryList.get(position);

        holder.nameText.setText(category.getName());
        holder.descriptionText.setText(category.getDescription());

        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(v.getContext(), EditCategoryActivity.class);
            intent.putExtra("id", category.getId());
            intent.putExtra("name", category.getName());
            intent.putExtra("description", category.getDescription());
            v.getContext().startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return categoryList.size();
    }

    static class CategoryViewHolder extends RecyclerView.ViewHolder {
        TextView nameText, descriptionText;

        public CategoryViewHolder(@NonNull View itemView) {
            super(itemView);
            nameText = itemView.findViewById(R.id.categoryNameText);
            descriptionText = itemView.findViewById(R.id.categoryDescriptionText);
        }
    }
}

File: app/src/main/java/com/example/e_commercemobapp/admin/AdminCategoryAdapter.kt
package com.example.e_commercemobapp.admin

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.e_commercemobapp.R
import com.example.e_commercemobapp.model.Category

class AdminCategoryAdapter(
    private val categories: MutableList<Category>,
    private val onClick: (Category) -> Unit
) : RecyclerView.Adapter<AdminCategoryAdapter.AdminCategoryViewHolder>() {

    inner class AdminCategoryViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val name: TextView = view.findViewById(R.id.categoryNameText)
        val desc: TextView = view.findViewById(R.id.categoryDescriptionText)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AdminCategoryViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.admin_item_category, parent, false)
        return AdminCategoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: AdminCategoryViewHolder, position: Int) {
        val cat = categories[position]

        holder.name.text = cat.name
        holder.desc.text = cat.description

        holder.itemView.setOnClickListener {
            onClick(cat)
        }
    }

    override fun getItemCount(): Int = categories.size
}
